### PR TITLE
ENG-18081 stop monitoring data sent if destination is down and exception is registered.

### PR DESCRIPTION
**Pull request by zeemanhe**

### CODE REVIEW

File: src/frontend/org/voltdb/rejoin/StreamSnapshotDataTarget.java [hunk 1]
```diff
@@ -320,17 +319,20 @@ public SnapshotSerializationException(String message) {
         final long m_bytesWrittenSinceConstruction;
         final long m_writeTimeout;
 
-        Watchdog(long bytesWritten, long writeTimout) {
+        // Last time aata written to destination
```
Comment (rdykiel) [inline src/frontend/org/voltdb/rejoin/StreamSnapshotDataTarget.java:322]: typo "aata"

File: src/frontend/org/voltdb/rejoin/StreamSnapshotDataTarget.java [hunk 2]
```diff
@@ -139,7 +138,7 @@ public StreamSnapshotDataTarget(long HSId, boolean lowestDestSite, Set<Long> all
                 CoreUtils.hsIdToString(HSId), m_targetId, (lowestDestSite?" [Lowest Site]":"")));
 
         // start a periodic task to look for timed out connections
-        VoltDB.instance().scheduleWork(new Watchdog(0, writeTimeout), WATCHDOG_PERIOS_S, -1, TimeUnit.SECONDS);
+        VoltDB.instance().scheduleWork(new Watchdog(0, writeTimeout, System.currentTimeMillis()), WATCHDOG_PERIOS_S, -1, TimeUnit.SECONDS);
```
Comment (rdykiel) [inline src/frontend/org/voltdb/rejoin/StreamSnapshotDataTarget.java:141]: Might want to fix typo in WATCHDOG_PERIOS_S -> WATCHDOG_PERIOD_S?

Comment (rdykiel) [inline src/frontend/org/voltdb/rejoin/StreamSnapshotDataTarget.java:141]: I was told many times that using nanoTime() was better to compute elapsed times

Comment (zeemanhe) [reply src/frontend/org/voltdb/rejoin/StreamSnapshotDataTarget.java:141]: use milliseconds here because timeout is tracked in millisecond.

**Pull request merged**